aten/src/ATen/native/Batching.cpp

```diff
@@ -20,6 +21,73 @@ static bool has_level(const Tensor& self, int64_t level) {
   return it != bdims.end();
 }
 
+// Returns a Tensor with batch dim with level `level` turned into a regular dimension,
+// as well as a logical dim index of where said dimension is in the returned tensor.
+//
+// The reason why we want to return the index of where said dimension is in the returned
+// tensor is because we want to keep track of which dimension used to be the batch
+// dimension so that we can move it to the correct logical dimension specified by
+// `out_dims` in vmap. For example, if we had
+// >>> x = torch.randn(2, 3, 5)
+// >>> vmap(lambda x: x, in_dims=0, out_dims=1)(x)
+// then right when we are about to exit the vmap block, x is a BatchedTensor with a
+// batch dimension at (physical) index 0. Note that the batch dimension doesn't
+// always have to exist at (physical) index 0. When we undo the batch dimension,
+// we want to move it to dimension 1 (as specified by out_dims). So we return the
+// index at which the batch dim appears so that we can move it to the correct place.
+// later down the line.
+static std::pair<Tensor,int64_t> remove_existing_batch_dim(const Tensor& self, int64_t level) {
+  const auto* batched = maybeGetBatched(self);
```

Comment:
minor nit: consider passing in `batched` as the argument, it lets you statically enforce the invariant, rather than a runtime test.

Reply:
Changed to take `batched` as argument